File: algorithms/sort/insertionSort/practice/InsertionSort1.java
import java.util.Arrays;

public class InsertionSort1 {
    public static void main(String[] args) {
        int[] unsortedArray = new int[] { 4, 1, 6, 8, 9, 2, 3, 0, 2 };
        insertionSort(unsortedArray);

        Arrays.stream(unsortedArray).forEach(System.out::println);
    }

    static void insertionSort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j = j - 1;
            }
            arr[j + 1] = key;
        }
        // ex. [2, 4, 1, ...]
    }
}

File: python/algorithms/sort/insertionSort/practice/insertion_sort_1.py
def insertion_sort(arr):
    for i in range(0, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1

        arr[j + 1] = key
    return arr

nums = [2, 5, 8, 3, 9, 4, 1]
print(insertion_sort(nums))
